### PR TITLE
Microphone input

### PR DESCRIPTION
**Pull request by andrewn**
This is very much a proof-of-concept.

The /microphone app on the device  listens to audio input and sends an FFT, db and gain across the WebSocket bridge. A UI at the external /microphone visualises this data.

screenshot 2018-01-17 18 55 52

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -36,6 +36,55 @@ Run the `deployment/provision` script from the command line.
 
     curl -sS get.pimoroni.com/phatdac | bash
 
+## USB Microphone
+
+These steps may vary based on your audio set-up. If you use a USB microphone:
```
Comment (andrewn) [inline INSTALL.md]: I don't really know how we deal with audio setup. It varies depending on the audio devices you choose to you and so is difficult to automate.

File: INSTALL.md [hunk 2]
```diff
@@ -36,6 +36,55 @@ Run the `deployment/provision` script from the command line.
...
+    Direct sample snooping device
+hw:CARD=Device,DEV=0
+    USB PnP Sound Device, USB Audio
+    Direct hardware device without any conversions
+plughw:CARD=Device,DEV=0
+    USB PnP Sound Device, USB Audio
+    Hardware device with all software conversions
+```
+
+In this instance we'll choose the last one `plughw:CARD=Device,DEV=0`, the "USB PnP Sound Device, USB Audio".
```
Comment (andrewn) [inline INSTALL.md]: I read somewhere that using these aliases might be more consistent than the `hw:0,0` approach but not sure.

File: services/manager/start-web-environment.sh [hunk 3]
```diff
@@ -4,7 +4,7 @@ INTERNAL_PORT=5001
 REMOTE_DEBUGGING_PORT=9222
 REMOTE_DEBUGGING_ADDRESS=0.0.0.0
 APP_URL=http://localhost:$INTERNAL_PORT/
-ARGS="--disable-gpu --remote-debugging-port=$REMOTE_DEBUGGING_PORT --remote-debugging-address=$REMOTE_DEBUGGING_ADDRESS $APP_URL"
+ARGS="--use-fake-ui-for-media-stream --disable-gpu --remote-debugging-port=$REMOTE_DEBUGGING_PORT --remote-debugging-address=$REMOTE_DEBUGGING_ADDRESS $APP_URL"
```
Comment (andrewn) [inline services/manager/start-web-environment.sh]: This disables the user permisson prompt.

File: INSTALL.md (hunk 2 above)
Comment (pixelblend) [reply INSTALL.md]: `0,0` just refers to the first audio device in the list. In order to ensure that your device matches that position, you had to juggle kernel modules (I think, it's been a while). This seems like a better approach.

**Comment by andrewn** [comment]
I'm closing this in favour of #104 as the app itself is useful but was in the old style. Also, we don't need the install documentation which was written for ALSA.

**Pull request closed without merging**